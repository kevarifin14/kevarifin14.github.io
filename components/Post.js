import ReactMarkdown from 'react-markdown';

import { CodeBlock, LinkRenderer } from 'utils';

export default function Post({ title, content }) {
  return (
    <div className="prose max-w-prose mx-auto prose-indigo py-8">

      <h1>{title}</h1>

      <ReactMarkdown
        escapeHtml={false}
        source={content}
        renderers={{ code: CodeBlock, link: LinkRenderer }}
      />
    </div>
  );
}
