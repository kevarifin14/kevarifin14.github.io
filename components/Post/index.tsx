import Markdown from 'components/Markdown';

type PostProps = {
  title: string,
  source: string,
}

export default function Post({ title, source }: PostProps) {
  return (
    <div className="prose mx-auto py-8">

      <h1>{title}</h1>

      <Markdown source={source} />

    </div>
  );
}
