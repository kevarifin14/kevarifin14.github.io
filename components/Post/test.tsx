import renderer from 'react-test-renderer';

import Post from '.';

describe('Post', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <Post />,
    );
    expect(tree).toMatchSnapshot();
  });
});
