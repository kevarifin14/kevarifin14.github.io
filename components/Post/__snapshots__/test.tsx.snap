// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches snapshot 1`] = `
<div
  className="prose max-w-prose mx-auto prose-indigo py-8"
>
  <h1 />
  <div
    className="max-w-prose w-full prose mx-auto"
  />
</div>
`;
